From 47707f5b358b2780a74d3f6da2562e9591276af0 Mon Sep 17 00:00:00 2001
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 24 Apr 2023 21:48:33 +0100
Subject: [PATCH 1/3] gnu: Add cl-chlorophyll.

* gnu/packages/lisp-xyz.scm (cl-chlorophyll, ecl-chlorophyll,
sbcl-chlorophyll): New variables.
---
 gnu/packages/lisp-xyz.scm | 31 +++++++++++++++++++++++++++++++
 1 file changed, 31 insertions(+)

diff --git a/gnu/packages/lisp-xyz.scm b/gnu/packages/lisp-xyz.scm
index d7b0531bcc..fcc549d632 100644
--- a/gnu/packages/lisp-xyz.scm
+++ b/gnu/packages/lisp-xyz.scm
@@ -23722,6 +23722,37 @@ (define-public cl-chemboy
 (define-public ecl-chemboy
   (sbcl-package->ecl-package sbcl-chemboy))
 
+(define-public sbcl-chlorophyll
+  (let ((commit "f2a50ad36d2fb11823c0e5a8da08a0de89c5c35f")
+        (revision "1"))
+    (package
+      (name "sbcl-chlorophyll")
+      (version (git-version "0.0.1" revision commit))
+      (source
+       (origin
+         (method git-fetch)
+         (uri (git-reference
+               (url "https://github.com/gorozhin/chlorophyll")
+               (commit commit)))
+         (file-name (git-file-name "chlorophyll" version))
+         (sha256
+          (base32 "1hgl8sjsmyqx4gs77q4p94b63zgpxk1wi9w9niki8j0213dr1s3y"))))
+      (build-system asdf-build-system/sbcl)
+      (native-inputs (list sbcl-fiveam))
+      (inputs (list sbcl-alexandria))
+      (home-page "https://github.com/gorozhin/chlorophyll")
+      (synopsis "ANSI escape code library for Common Lisp")
+      (description
+"This package provides a Common Lisp system CHLOROPHYLL which implements an
+ANSI escape code functionality.")
+      (license license:expat))))
+
+(define-public cl-chlorophyll
+  (sbcl-package->cl-source-package sbcl-chlorophyll))
+
+(define-public ecl-chlorophyll
+  (sbcl-package->ecl-package sbcl-chlorophyll))
+
 (define-public sbcl-cl-pass
   (let ((commit "e58e97c0c0588dc742c061208afb9bc31e4dbd34")
         (revision "1"))
-- 
2.39.2

